test/BinaryParse.cpp

```diff
@@ -24,12 +24,428 @@
+// Assembler generator, and with a given Id bound.
+Words ExpectedHeaderForBound(uint32_t bound) {
+  return {SpvMagicNumber, SpvVersion,
+          SPV_GENERATOR_WORD(SPV_GENERATOR_KHRONOS_ASSEMBLER, 0), bound, 0};
+}
+
+// Returns a parsed operand for a non-number value at the given word offset
+// within an instruction.
+spv_parsed_operand_t MakeSimpleOperand(uint16_t offset,
+                                       spv_operand_type_t type) {
+  return {offset, 1, type, SPV_NUMBER_NONE, 0};
+}
+
+// Returns a parsed operand for a literal unsigned integer value at the given
+// word offset within an instruction.
+spv_parsed_operand_t MakeLiteralNumberOperand(uint16_t offset) {
+  return {offset, 1, SPV_OPERAND_TYPE_LITERAL_INTEGER, SPV_NUMBER_UNSIGNED_INT,
+          32};
+}
+
+// Returns a parsed operand for a literal string value at the given
+// word offset within an instruction.
+spv_parsed_operand_t MakeLiteralStringOperand(uint16_t offset,
+                                              uint16_t length) {
+  return {offset, length, SPV_OPERAND_TYPE_LITERAL_STRING, SPV_NUMBER_NONE, 0};
+}
+
+// Returns a ParsedInstruction for an OpTypeVoid instruction that would
+// generate the given result Id.
+ParsedInstruction MakeParsedVoidTypeInstruction(uint32_t result_id) {
+  const auto void_inst = MakeInstruction(SpvOpTypeVoid, {result_id});
+  const auto void_operands = std::vector<spv_parsed_operand_t>{
+      MakeSimpleOperand(1, SPV_OPERAND_TYPE_RESULT_ID)};
+  const spv_parsed_instruction_t parsed_void_inst = {
+      void_inst.data(),
+      uint16_t(void_inst.size()),
+      SpvOpTypeVoid,
+      SPV_EXT_INST_TYPE_NONE,
+      0,  // type id
+      result_id,
+      void_operands.data(),
+      uint16_t(void_operands.size())};
+  return ParsedInstruction(parsed_void_inst);
+}
+
+// Returns a ParsedInstruction for an OpTypeInt instruction that generates
+// the given result Id for a 32-bit signed integer scalar type.
+ParsedInstruction MakeParsedInt32TypeInstruction(uint32_t result_id) {
+  const auto i32_inst = MakeInstruction(SpvOpTypeInt, {result_id, 32, 1});
+  const auto i32_operands = std::vector<spv_parsed_operand_t>{
+      MakeSimpleOperand(1, SPV_OPERAND_TYPE_RESULT_ID),
+      MakeLiteralNumberOperand(2), MakeLiteralNumberOperand(3)};
+  spv_parsed_instruction_t parsed_i32_inst = {i32_inst.data(),
+                                              uint16_t(i32_inst.size()),
+                                              SpvOpTypeInt,
+                                              SPV_EXT_INST_TYPE_NONE,
+                                              0,  // type id
+                                              result_id,
+                                              i32_operands.data(),
+                                              uint16_t(i32_operands.size())};
+  return ParsedInstruction(parsed_i32_inst);
+}
+
+using BinaryParseTest =
+    spvtest::TextToBinaryTestBase<CaptureParseResults<::testing::Test>>;
```

Comment:
I believe the current way leads to less code and more readable tests, once you've digested this using definition.
Otherwise you have to carry around the capturer object and give it a name.

(And before it's suggested, every test explicitly writes out the spvBinaryParse call, as per advice I absorbed from
https://cppcon2015.sched.org/event/3vmd/your-tests-are-terrible-tales-from-the-trenches)